core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1010,6 +1010,65 @@ private[spark] object Utils extends Logging {
     )
   }
 
+  /** Check whether a time-suffix was provided for the time string. */
+  private def hasTimeSuffix(str: String) : Boolean = {
+    val lower = str.toLowerCase.trim()
+    lower.endsWith("ms") || lower.endsWith("us") || lower.endsWith("s")
+  }
+
+  val timeError = "Time must be specified as seconds (s), " +
+      "milliseconds (ms), or microseconds (us) e.g. 50s, 100ms, or 250us."
+
+  /**
+   * Convert a passed time string (e.g. 50s, 100ms, or 250us) to a microsecond count for
+   * internal use. If no suffix is provided a direct conversion is attempted.
+   */
+  @throws(classOf[NumberFormatException])
+  private def timeStringToUs(str: String) : Long = {
+    try {
+      val lower = str.toLowerCase.trim()
+      if (lower.endsWith("ms")) {
```

Comment:
I'd add at least `min`, `h` and `d` as valid suffixes here. Also, something like this might be easier to code for:

    val suffixes = Map(
      "us" -> 1,
      "ms" -> 1000,
      "s" -> 1000 * 1000,
      "min" -> 60 * 1000 * 1000,
      "h" -> 60 * 60 * 1000 * 1000,
      // etc

Then you can iterate over the map instead of having cascading if / else checks.